Mathlib/RepresentationTheory/GroupCohomology/Hilbert90.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2023 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.FieldTheory.Galois
+import Mathlib.RepresentationTheory.GroupCohomology.LowDegree
+import Mathlib.LinearAlgebra.LinearIndependent
+
+/-!
+# Hilbert's Theorem 90
+
+Let `L/K` be a finite extension of fields. Then this file proves Noether's generalization of
+Hilbert's Theorem 90: that the 1st group cohomology $H^1(Aut_K(L), Lˣ)$ is trivial. We state it
+both in terms of $H^1$ and otherwise.
+
+Hilbert's original statement was that if $L/K$ is Galois, and $Gal(L/K)$ is cyclic, generated
+by an element `σ`, then for every `x : L` such that $N_{L/K}(x) = 1,$ there exists `y : L` such
+that $x = y/σ(y).$ This can be deduced from the fact that the function $Gal(L/K) → L^\times$
+sending $σ^i \mapsto xσ(x)σ^2(x)...σ^{i-1}(x)$ is a 1-cocycle. Alternatively, we can derive it by
+analyzing the cohomology of finite cyclic groups in general.
+
+Noether's generalization also holds for infinite Galois extensions.
+
+## Main definitions
+
+* `hilbert90`: for all $f: Aut_K(L) \to L^\times$ satisfying the 1-cocycle condition, there exists
+`β : Lˣ` such that $f(g)g(β) = β$ for all `g : Aut_K(L)`.
+* `groupCohomology.hilbert90`: $H^1(Aut_K(L), L^\times)$ is trivial.
+
+## Implementation notes
+
+Given a commutative ring `k` and a group `G`, group cohomology is developed in terms of `k`-linear
+`G`-representations on `k`-modules. Thus stating Noether's generalization of Hilbert 90 in terms
+of `H¹` requires us to turn the natural action of `Aut_K(L)` on `Lˣ` into a morphism
+`Aut_K(L) →* (Additive Lˣ →ₗ[ℤ] Additive Lˣ)`. Thus we provide the non-`H¹` version too, as its
+statement is clearer.
+
+## TODO
+
+* The original Hilbert's Theorem 90, deduced from the cohomology of general finite cyclic groups.
+* Develop Galois cohomology to extend Noether's result to infinite Galois extensions.
+* "Additive Hilbert 90": let `L/K` be a finite Galois extension. Then $H^n(Gal(L/K), L)$ is trivial
+for all $1 ≤ n.$
+
+-/
+suppress_compilation
+
+open BigOperators
+namespace Hilbert90
+
+variable (K L : Type*) [Field K] [Field L] [Algebra K L] [FiniteDimensional K L]
+
+/-- Given `f : Aut_K(L) → Lˣ`, the sum `∑ f(φ) • φ` for `φ ∈ Aut_K(L)`, as a function `L → L`. -/
+noncomputable def aux (f : (L ≃ₐ[K] L) → Lˣ) : L → L :=
+  Finsupp.total (L ≃ₐ[K] L) (L → L) L (fun φ => φ)
+    (Finsupp.equivFunOnFinite.symm (fun φ => (f φ : L)))
+
+theorem aux_ne_zero (f : (L ≃ₐ[K] L) → Lˣ) : aux K L f ≠ 0 :=
+/- the set `Aut_K(L)` is linearly independent in the `L`-vector space `L → L`, by Dedekind's
+linear independence of characters -/
+  have : @LinearIndependent (L ≃ₐ[K] L) L (L → L) (fun f => f) _ _ _ :=
```

Comment:
Can you use `(... := ...)` instead of `@` here? It is usually more readable.

Reply:
thanks, done